[webkitpy] Add VisionOSDevicePort.

bls1999 opened the pull request:
#### 588c4da089ce18071c3f0685c3380fb54476e413
<pre>
[webkitpy] Add VisionOSDevicePort.
<a href="https://bugs.webkit.org/show_bug.cgi?id=293140">https://bugs.webkit.org/show_bug.cgi?id=293140</a>
<a href="https://rdar.apple.com/151479231">rdar://151479231</a>

Reviewed by Sam Sneddon.

This PR adds a port for visionOS devices, as Vision is the only simulated
device type that doesn&apos;t have a matching device port.

* Tools/Scripts/webkitpy/common/system/platforminfo.py:
    (PlatformInfo):
        -&gt; (is_visionos): Returns true if the platform is visionOS.
        -&gt; (_determine_os_name): Adds a switch case that &apos;visionos&apos; if the platform is visionOS.
* Tools/Scripts/webkitpy/port/factory.py:
    (PortFactory):
        -&gt; (PORT_CLASSES): Add VisionOSDevicePort.
* Tools/Scripts/webkitpy/port/visionos.py:
    (VisionOSPort):
        -&gt; (CURRENT_VERSION): Bump to 2.
        -&gt; (default_baseline_search_path): Differentiate between simulator and device.
* Tools/Scripts/webkitpy/port/visionos_device.py: Added.
* Tools/Scripts/webkitpy/xcode/device_type.py:
    (DeviceType):
        -&gt; (_define_software_variant_from_hardware_family): Add handling for visionOS that is consistent with other platforms.
        -&gt; (check_consistency): Ditto.

Canonical link: <a href="https://commits.webkit.org/295033@main">https://commits.webkit.org/295033@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0e6e56f6e96afe830a636963361a6516fefac2fe

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/103851 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23554 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/13875 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109044 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54512 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/105891 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/23906 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32099 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/78902 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/106857 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18561 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/93685 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/59231 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/103327 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/18360 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/11735 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/53879 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/88121 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/11794 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111431 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31007 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/22864 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/87903 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31369 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/89886 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87554 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32445 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/10176 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/25372 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16860 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/30936 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/30729 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34066 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/32290 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/webkitpy/port/visionos.py:
```diff
@@ -55,7 +55,10 @@ def version_name(self):
         return VersionNameMap.map(self.host.platform).to_name(self._os_version, platform=VisionOSPort.port_name)
 
-    def default_baseline_search_path(self, **kwargs):
+    def default_baseline_search_path(self, device_type=None, **kwargs):
```
Do we need this change for visionOS? This seems more related to iOS.

bls1999 replied:
Yeah, this allows device-specific baseline mapping.

gsnedders reviewed (approved):
I still wish we could do better at unifying more of all of the Apple embedded platform ports, but I don't think it makes sense to block this PR on that.

bls1999 committed:
[webkitpy] Add VisionOSDevicePort.
https://bugs.webkit.org/show_bug.cgi?id=293140
rdar://151479231

Reviewed by Sam Sneddon.

This PR adds a port for visionOS devices, as Vision is the only simulated
device type that doesn't have a matching device port.

* Tools/Scripts/webkitpy/common/system/platforminfo.py:
    (PlatformInfo):
        -> (is_visionos): Returns true if the platform is visionOS.
        -> (_determine_os_name): Adds a switch case that 'visionos' if the platform is visionOS.
* Tools/Scripts/webkitpy/port/factory.py:
    (PortFactory):
        -> (PORT_CLASSES): Add VisionOSDevicePort.
* Tools/Scripts/webkitpy/port/visionos.py:
    (VisionOSPort):
        -> (CURRENT_VERSION): Bump to 2.
        -> (default_baseline_search_path): Differentiate between simulator and device.
* Tools/Scripts/webkitpy/port/visionos_device.py: Added.
* Tools/Scripts/webkitpy/xcode/device_type.py:
    (DeviceType):
        -> (_define_software_variant_from_hardware_family): Add handling for visionOS that is consistent with other platforms.
        -> (check_consistency): Ditto.

Canonical link: https://commits.webkit.org/295033@main
